srv/src/main/java/my/bookshop/handlers/SingletonInitialization.java

```diff
@@ -0,0 +1,29 @@
+package my.bookshop.handlers;
+
+import com.sap.cds.ql.Upsert;
+import com.sap.cds.services.application.ApplicationLifecycleService;
+import com.sap.cds.services.handler.EventHandler;
+import com.sap.cds.services.handler.annotations.After;
+import com.sap.cds.services.handler.annotations.ServiceName;
+import com.sap.cds.services.persistence.PersistenceService;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import cds.gen.my.bookshop.Bookshop_;
+import cds.gen.my.bookshop.Csv;
+
+@Component
+@ServiceName(ApplicationLifecycleService.DEFAULT_NAME)
+public class SingletonInitialization implements EventHandler {
+
+	@Autowired
+	private PersistenceService db;
+	
+	@After(event = ApplicationLifecycleService.EVENT_APPLICATION_PREPARED)
+	public void initializeCsvEntity() {
+		Csv csv = Csv.create();
+		csv.setDataType("text/csv");
+		db.run(Upsert.into(Bookshop_.CSV).entry(csv));
```

Comment:
This approach is a little problematic with a multi-tenant application. Most likely it will also break the multi-tenant scenario, as in that case it is not clear in which database container to insert the data.

What about doing that via CSV initialization files as well? They get deployed on every tenant database during tenant onboarding.

Alternatively we could also try to implement this fully without a database persistence (which is not really required). Most likely we just need to return a static Csv entity for all READ events on this entity on the Admin Service.